tensorboard/plugins/histogram/summary.py

```diff
@@ -124,27 +128,54 @@ def op(name,
 
   Returns:
     A TensorFlow summary op.
+
+  Raises:
+    ValueError: If data is not rank 1 when `edges` is specified.
+    ValueError: If the shape of data is not [bucket_count] or edges is not
+      [bucket_count+1].
   """
   if display_name is None:
     display_name = name
   summary_metadata = metadata.create_summary_metadata(
       display_name=display_name, description=description)
   with tf.name_scope(name):
-    tensor = _buckets(data, bucket_count=bucket_count)
-    return tf.summary.tensor_summary(name='histogram_summary',
-                                     tensor=tensor,
-                                     collections=collections,
-                                     summary_metadata=summary_metadata)
-
-
-def pb(name, data, bucket_count=None, display_name=None, description=None):
+    if edges is None:
+      tensor = _buckets(data, bucket_count=bucket_count)
+    else:
+      if len(data.get_shape()) != 1 or len(edges.get_shape()) != 1:
```

Comment:
This code doesn't work with dynamic shapes. (In fact, it will hard-error if `edges.get_shape()` is `TensorShape(None)`.) We'd like all plugins to accept dynamic shapes whenever possible.\*

For the rank check: [`tf.assert_rank`].

For the shape check: you want `tf.shape(tensor)`, which gives you a tensor containing the shape data; then, you can use a `tf.assert_*` with control dependencies.

[`tf.assert_rank`]: https://www.tensorflow.org/versions/master/api_docs/python/tf/assert_rank

\* Here, "dynamic" means "not known to the TensorFlow compiler, which is kind of smart but not omniscient." Shapes can be lost in translation in various points of normal use. We have tests for this:  
https://github.com/tensorflow/tensorboard/blob/dad4df940deaf5b94751c125aeea1cee56c67522/tensorboard/plugins/histogram/summary_test.py#L130-L136